file.c

```diff
@@ -533,7 +533,10 @@ filewclose(File *f)
     if (!f) return;
     if (!f->iswopen) return;
     if (f->free) {
-        (void)ftruncate(f->fd, f->w->filesize - f->free);
+        errno = 0;
+        if (ftruncate(f->fd, f->w->filesize - f->free) != 0) {
+            twarn("ftruncate failed, errno=%d", errno);
```

Comment:
Note that twarn mostly just calls the stdlib function warn, which has errno/strerror built in.

```
The err(), errc(), verr(), verrc(), warn(), warnc(), vwarn(), and
vwarnc() functions append an error message obtained from strerror(3)
based on a supplied error code value or the global variable errno, pre-
ceded by another colon and space unless the fmt argument is NULL.
```

So I think this line would print about the same information if it were written like this:

```c
twarn("ftruncate"); 
```

Reply:
Thanks!